src/Stoa.ts

```diff
@@ -1704,6 +1711,136 @@ class Stoa extends WebService
+                    circulating_supply: 5283535,
+                    active_validators: 155055
+                };
+            this.socket.io.emit(events.server.lateststats, boaStats)
+            logger.info(`Emited Updated BOA stats:  ${boaStats}`)
+            return resolve(boaStats);
+            }
+        })
+        .catch((err) => {
+            logger.error("Failed to latest BOA stats: " + err);
+           return;
+        });
+        });
+    }
+
+    /**
+     *  Stoa emits the updates using sockets on new block recieved
+     * @param height The height of block to emit  
+     * @returns Returns the Promise. If it is finished successfully the `.then`
+     * of the returned Promise is called
+     * and if an error occurs the `.catch` is called with an error.
+     */
+    public emitBlock(height: Height): Promise<boolean> {
+        return new Promise<boolean>(async (resolve, reject) => {
+            try{
+               let emittedBlock :IBlock = await this.emitNewBlock(height);
+               await this.emitBlockTransactions(emittedBlock)
+               resolve(true);
+             }
+            catch(err)
+            {
+                reject("Failed to emit new block")
+            }
+        });
+    }
+
+    /**
+     * Stoa emits the detail of new block received
+     * @param height  
+     * @returns 
+     */
+    public emitNewBlock(height: Height): Promise<IBlock> {
+        return new Promise<IBlock>((resolve, reject) => {
+            this.ledger_storage.getBlock(height).then((data) => {
+                if (data.length === undefined || data.length === 0) {
+                    return  reject("Failed to emit new block")
+                }
+                else {
+                    let latestBlock: IBlock = {
+                        height: JSBI.BigInt(data[0].height).toString(),
+                        hash: new Hash(data[0].hash, Endian.Little).toString(),
+                        merkle_root: new Hash(data[0].merkle_root, Endian.Little).toString(),
+                        signature: new Hash(data[0].signature, Endian.Little).toString(),
+                        validators: data[0].validators.toString(),
+                        tx_count: (data[0].tx_count).toString(),
+                        enrollment_count: data[0].enrollment_count.toString(),
+                        time_stamp: data[0].time_stamp,
+                    }
+                    logger.info(`Emited new Block: ${latestBlock}`)
+                    this.socket.io.emit(events.server.newblock, latestBlock)
+                    return resolve(latestBlock);
+                }
+            }).catch((err) => {
+                return reject("Failed to emit new block")
+            });
+        });
+    }
+    /**
+     * Stoa emit the transaction inside the new block recieved.
+     * @param block 
+     * @returns 
+     */
+    public emitBlockTransactions(block: IBlock): Promise<IBlockTransactions> {
```

Comment:
It would be better to emit the data with the interface as below.
``` TypeScrype
export interface IEmitTransaction
{
    /**
     * block height
     */
    height: string;

    /**
     * hash of block
     */
    hash: string;
    
    /**
     * hash of transaction
     */
    tx_hash: string;

    transaction: Transaction;
}
```

The interface member `transaction` can use the data received from Agora as it is.
It would be better to emit it immediately without querying it in the database.